libs/coin-framework/src/api/types.ts

```diff
@@ -321,20 +321,47 @@ type MaybeTxData<TxDataType extends TxData> = TxDataType extends TxDataNotSuppor
 
 export type FeesStrategy = "slow" | "medium" | "fast";
 
+export type StakingOperation =
+  | "delegate"
+  | "undelegate"
+  | "redelegate"
+  | "getStakedBalance"
+  | "getUnstakedBalance";
+
 export type TransactionIntent<
   MemoType extends Memo = MemoNotSupported,
   TxDataType extends TxData = TxDataNotSupported,
 > = {
+  intentType: "transaction" | "staking";
   type: string;
   sender: string;
-  senderPublicKey?: string;
-  expiration?: number;
   recipient: string;
   amount: bigint;
-  useAllAmount?: boolean;
   asset: AssetInfo;
-  sequence?: number;
+  useAllAmount?: boolean;
```

Comment:
Similar comment between `amount` and `useAllAmount`, maybe `amount: bigint | "all"`?